### ISSUE TITLE
Will not build with gcc7.1

### ISSUE THREAD

**Issue by JesApp** [rawrtc/rawrtc#68]
The dependency usrsctp does not build with gcc7.1. Updating the dependency makes rawrtc not build anymore.

*The problem*

Running the `make-dependencies.sh` script with gcc version 7.1.1 fails during the building of  [rawrtc/usrsctp](https://github.com/rawrtc/usrsctp) with several warnings of type `format-truncation` (which become errors thanks to `-Werror`). 

*Possible solution*

This was documented and fixed [upstream](https://github.com/sctplab/usrsctp/issues/149). Using the `master` branch of [sctplab/usrsctp](https://github.com/sctplab/usrsctp/tree/f4819e1b177f7bfdd761c147f5a649b9f1a78c06) makes `make-dependencies.sh` run without errors, but then creates errors when building rawrtc itself.

A good solution would be to make rawrtc work with a more recent version of usrsctp instead of rawrtcs own fork, however there is no actual release that contains the changes needed to build with gcc7.1 (see [this issue](https://github.com/sctplab/usrsctp/issues/136)).

*Workaround*

Unsurprisingly, everything works fine with gcc 5.4.0.

*Side Note*

Using clang instead of any gcc creates it's own selection of errors.

**Comment by lgrahl** [comment]
That requires bringing the upstream changes from [sctplab/usrsctp:master](https://github.com/sctplab/usrsctp/tree/master) into [NEAT-project/usrsctp-neat:usrsctpForNeat](https://github.com/NEAT-project/usrsctp-neat/tree/usrsctpForNeat) and I'm not maintaining that project (ping to @ruengeler, @tuexen or @weinrank). Once this is complete, I'll happily pull the upstream changes and update RAWRTC to work with gcc7.1.

Please open a separate issue for the clang errors including the console output and the specific OS and clang version.

Note: The fork only exists to pin a specific commit of usrsctp-neat to RAWRTC. I'd be happy to remove the fork but it requires way more frequent releases from usrsctp-neat. (The same can actually be said about this project... will release a version soon and from then on bump the version regularly.)

**Comment by tuexen** [comment]
I'm not maintaining [usrsctpForNeat](https://github.com/NEAT-project/usrsctp-neat/tree/usrsctpForNeat), this is done by @ruengeler  or @weinrank. My understanding is that [usrsctpForNeat](https://github.com/NEAT-project/usrsctp-neat/tree/usrsctpForNeat) is kept up to date with [usrsctp](https://github.com/sctplab/usrsctp/tree/master) and contains additional stuff eventually being integrated into [usrsctp](https://github.com/sctplab/usrsctp/tree/master).

**Comment by lgrahl** [comment]
Okay, thanks for the clarification. RAWRTC will of course also go back to sctplab/usrsctp once all the work from usrsctp-neat has been integrated back into usrsctp.

**Comment by tuexen** [comment]
@lgrahl Makes sense.

**Comment by dbrgn** [comment]
I'm also facing this issue.

As a quick hack, to get compilation working again:

```diff
diff --git a/make-dependencies.sh b/make-dependencies.sh
index 568fd34..eee54ea 100755
--- a/make-dependencies.sh
+++ b/make-dependencies.sh
@@ -178,6 +178,7 @@ cmake -DCMAKE_INSTALL_PREFIX=${PREFIX} -DSCTP_DEBUG=1 ..
 echo "Cleaning usrsctp"
 make clean
 echo "Building & installing usrsctp"
+sed -i 's/SCTP_DIAG_INFO_LEN 64/SCTP_DIAG_INFO_LEN 128/' ../usrsctplib/netinet/sctp_constants.h
 make install -j${THREADS}
 rm -f ${PREFIX}/lib/libusrsctp.so* ${PREFIX}/lib/libusrsctp.*dylib
 cd ${MAIN_DIR}
```

**Comment by lgrahl** [comment]
Moved to https://github.com/rawrtc/rawrtc-data-channel/issues/2

**Comment by mo3rfan** [comment]
I had to increase the SCTP_DIAG_INFO_LEN to 136 instead of 128 to compile on GCC 8.1
